airbyte-integrations/connectors/destination-s3/src/main/java/io/airbyte/integrations/destination/s3/S3DestinationConfig.java

```diff
@@ -126,15 +133,32 @@ public S3FormatConfig getFormatConfig() {
     return formatConfig;
   }
 
-  public AmazonS3 getS3Client() {
+  public synchronized AmazonS3 getS3Client() {
```

Comment:
nitpick: I generally avoid `synchronized` methods in favor of just having a `private Object LOCK = new Object();` field, and using `synchronized (LOCK) { ... }`, just in case anyone else uses one of these `S3DestinationConfig` objects as their own semaphore